common/static/common/js/discussion/views/discussion_content_view.js

```diff
@@ -366,41 +366,41 @@
                 event.preventDefault();
-                is_endorsing = !this.model.get("endorsed");
-                url = this.model.urlFor("endorse");
+                is_endorsing = !this.model.get('endorsed');
+                url = this.model.urlFor('endorse');
                 updates = {
                     endorsed: is_endorsing,
                     endorsement: is_endorsing ? {
-                        username: DiscussionUtil.getUser().get("username"),
+                        username: DiscussionUtil.getUser().get('username'),
                         user_id: DiscussionUtil.getUser().id,
                         time: new Date().toISOString()
                     } : null
                 };
                 if (this.model.get('thread').get('thread_type') === 'question') {
                     if (is_endorsing) {
-                        msg = gettext("We had some trouble marking this response as an answer.  Please try again.");
+                        msg = gettext('We had some trouble marking this response as an answer.  Please try again.');
                     } else {
-                        msg = gettext("We had some trouble removing this response as an answer.  Please try again.");
+                        msg = gettext('We had some trouble removing this response as an answer.  Please try again.');
                     }
                 } else {
                     if (is_endorsing) {
-                        msg = gettext("We had some trouble marking this response endorsed.  Please try again.");
+                        msg = gettext('We had some trouble marking this response endorsed.  Please try again.');
                     } else {
-                        msg = gettext("We had some trouble removing this endorsement.  Please try again.");
+                        msg = gettext('We had some trouble removing this endorsement.  Please try again.');
                     }
                 }
-                beforeFunc = function() {
-                    return self.trigger("comment:endorse");
-                };
-                return DiscussionUtil.updateWithUndo(this.model, updates, {
-                    url: url,
-                    type: "POST",
-                    data: {
-                        endorsed: is_endorsing
+                return DiscussionUtil.updateWithUndo(
+                    this.model,
+                    updates,
+                    {
+                        url: url,
+                        type: 'POST',
+                        data: {endorsed: is_endorsing},
+                        $elem: $(event.currentTarget)
                     },
-                    beforeSend: beforeFunc,
-                    $elem: $(event.currentTarget)
-                }, msg).always(this.trigger("comment:endorse"));
+                    msg,
+                    function() { return self.trigger('comment:endorse'); }
+                ).always(this.trigger('comment:endorse'));
```

Comment:
Ah good catch!  251f758 merges upstream's discussion/utils.js, and adds the 5th `beforeSend` parameter.